test/e2e/network/endpointslice.go

```diff
@@ -524,6 +526,209 @@ var _ = common.SIGDescribe("EndpointSlice", func() {
+				Name: "example-custom-endpoints",
+			},
+			Spec: v1.ServiceSpec{
+				Ports: []v1.ServicePort{
+					{
+						Name:     "port80",
+						Port:     80,
+						Protocol: v1.ProtocolTCP,
+					},
+					{
+						Name:     "port81",
+						Port:     81,
+						Protocol: v1.ProtocolTCP,
+					},
+				},
+			},
+		})
+
+		// Add a backend pod to the service in the other node
+		port8090 := []v1.ContainerPort{
+			{
+				ContainerPort: 8090,
+				Protocol:      v1.ProtocolTCP,
+			},
+		}
+		port9090 := []v1.ContainerPort{
+			{
+				ContainerPort: 9090,
+				Protocol:      v1.ProtocolTCP,
+			},
+		}
+
+		serverPod := e2epod.NewAgnhostPodFromContainers(
+			"", "pod-handle-http-request", nil,
+			e2epod.NewAgnhostContainer("container-handle-8090-request", nil, port8090, "netexec", "--http-port", "8090", "--udp-port", "-1"),
+			e2epod.NewAgnhostContainer("container-handle-9090-request", nil, port9090, "netexec", "--http-port", "9090", "--udp-port", "-1"),
+		)
+
+		pod := e2epod.NewPodClient(f).CreateSync(serverPod)
+
+		if pod.Status.PodIP == "" {
+			framework.Failf("PodIP not assigned for pod %s", pod.Name)
+		}
+
+		addresType := discoveryv1.AddressTypeIPv4
+		if framework.TestContext.ClusterIsIPv6() {
+			addresType = discoveryv1.AddressTypeIPv6
+		}
+
+		// create custom endpoint slices
+		tcpProtocol := v1.ProtocolTCP
+		epsTemplate := &discoveryv1.EndpointSlice{
+			ObjectMeta: metav1.ObjectMeta{GenerateName: "e2e-custom-slice",
+				Labels: map[string]string{
+					discoveryv1.LabelServiceName: svc.Name,
+					discoveryv1.LabelManagedBy:   "e2e-test" + ns,
+				}},
+			AddressType: addresType,
+			Endpoints: []discoveryv1.Endpoint{
+				{Addresses: []string{pod.Status.PodIP}},
+			},
+		}
+
+		ginkgo.By("creating")
+		eps1 := epsTemplate.DeepCopy()
+		eps1.Ports = []discoveryv1.EndpointPort{{
+			Name:     pointer.String("port80"),
+			Port:     pointer.Int32(8090),
+			Protocol: &tcpProtocol,
+		}}
+
+		_, err := f.ClientSet.DiscoveryV1().EndpointSlices(ns).Create(context.TODO(), eps1, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		eps2 := epsTemplate.DeepCopy()
+		eps2.Ports = []discoveryv1.EndpointPort{{
+			Name:     pointer.String("port81"),
+			Port:     pointer.Int32(9090),
+			Protocol: &tcpProtocol,
+		}}
+
+		_, err = f.ClientSet.DiscoveryV1().EndpointSlices(ns).Create(context.TODO(), eps2, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		// connect to the service must work
+		ginkgo.By("Creating a pause pods that will try to connect to the webserver")
+		pausePod0 := e2epod.NewAgnhostPod(ns, "pause-pod-0", nil, nil, nil)
+		e2epod.NewPodClient(f).CreateSync(pausePod0)
+
+		dest1 := net.JoinHostPort(svc.Spec.ClusterIP, "80")
+		dest2 := net.JoinHostPort(svc.Spec.ClusterIP, "81")
+		execHostnameTest(*pausePod0, dest1, serverPod.Name)
+		execHostnameTest(*pausePod0, dest2, serverPod.Name)
+
+	})
+
+	ginkgo.It("should forward traffic to a Service with custom endpoints in different EndpointSlices", func() {
```

Comment:
`"should support a Service with multiple endpoint IPs specified in multiple EndpointSlices"`

Reply:
done